beautyjoy/BJC-Teacher-Tracker
# CS169L SP24: Iter 2-4 Changes

## This PR incorporates many changes

- tooltip bug fix on admin tables
- multiple email addresses per teach
- languages for teachers
- PD model
- pd registrations

----

### [Pivotal Tracker Link][tracker]


[tracker]: https://www.pivotaltracker.com/story/show/187460464

## What this PR does:


This pull request addresses a bug related to tooltip persistence upon clicking checkboxes. Previously, tooltips remained visible even after clicking a checkbox. The solution involves triggering a redraw of the table with `$tables.draw()` upon a checkbox change. Additionally, during each table redraw, all tooltips are first disposed of and then reinitialized. This ensures that tooltips only appear when needed and do not persist inappropriately.

### Include screenshots, videos, etc.
Before:
<img width="473" alt="Screenshot 2024-04-25 at 17 23 50" src="https://github.com/beautyjoy/BJC-Teacher-Tracker/assets/90062350/607b2430-b62d-431c-89da-8b16c78fa4a8">
After:
<img width="425" alt="Screenshot 2024-04-25 at 17 23 55" src="https://github.com/beautyjoy/BJC-Teacher-Tracker/assets/90062350/336ccd37-3a64-4508-8ea0-bea4b954a6ea">


#### Who authored this PR?
@perryzjc 
@realmichaeltao 

### How should this PR be tested?

* Is there a deploy we can view? 
Yes
* What do the specs/features test?
Since this change primarily affects frontend functionality, traditional backend testing frameworks like Cucumber or RSpec are not applicable here.
* Are there edge cases to watch out for?
Check tooltip behavior when multiple checkboxes are interacted with in quick succession.

#### Are there any complications to deploying this? 
No

### Checklist:
